backend/src/main/java/com/github/jekattack/cornergame/kioskdata/KioskService.java

```diff
@@ -32,4 +34,20 @@ public void createManyKiosks(List<Kiosk> kioskList) {
         });
     }
 
+    public ActiveQuestDTO getLocationsForQuest(String[] questPlacesIds) {
+        List<ActiveQuestWaypointDTO> waypoints = new ArrayList<>();
+        KioskLocationCoordinates start = new KioskLocationCoordinates();
+        KioskLocationCoordinates finish = new KioskLocationCoordinates();
+        for(int i=0; i<questPlacesIds.length; i++){
+            if(i==0){
+                start = kioskRepository.findByGooglePlacesId(questPlacesIds[i]).stream().findFirst().orElseThrow().getKioskLocation().getLocation();
+            } else if(i==(questPlacesIds.length-1)) {
+                finish = kioskRepository.findByGooglePlacesId(questPlacesIds[i]).stream().findFirst().orElseThrow().getKioskLocation().getLocation();
+            } else {
+                waypoints.add(new ActiveQuestWaypointDTO(kioskRepository.findByGooglePlacesId(questPlacesIds[i]).stream().findFirst().orElseThrow().getKioskLocation().getLocation()));
+            }
+        }
+
+        return new ActiveQuestDTO(start, waypoints, finish);
+    }
```

Comment:
This method can be optimized.
```java
public ActiveQuestDTO getLocationsForQuest(String[] questPlacesIds) {
    List<ActiveQuestWaypointDTO> waypoints = new ArrayList<>();
    KioskLocationCoordinates start = kioskRepository.findByGooglePlacesId(questPlacesIds[i]).stream().findFirst().orElseThrow().getKioskLocation().getLocation();
    KioskLocationCoordinates finish = kioskRepository.findByGooglePlacesId(questPlacesIds[i]).stream().findFirst().orElseThrow().getKioskLocation().getLocation();
    for(int i=1; i<questPlacesIds.length - 1; i++){
        waypoints.add(new ActiveQuestWaypointDTO(kioskRepository.findByGooglePlacesId(questPlacesIds[i]).stream().findFirst().orElseThrow().getKioskLocation().getLocation()));
    }

    return new ActiveQuestDTO(start, waypoints, finish);
}
```
That way you don't instantiate the two `KioskLocationCoordinates` instances.